.github/workflows/codeql.yaml

```
@@ -0,0 +1,43 @@
+---
+name: codeQL
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  analyze:
+    name: Static analysis with CodeQL
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    env:
+      NEXUS_UNAME: ${{ secrets.NEXUS_USERNAME }}
+      NEXUS_PWD: ${{ secrets.NEXUS_PASSWORD }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: "corretto"
+          java-version: "17"
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2.4.2
+        with:
+          gradle-version: 7.3
+      
+      - name: Execute build with Gradle wrapper
+        run: ./gradlew build -PsonatypeUsername=$NEXUS_UNAME -PsonatypePassword=$NEXUS_PWD
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: java
+
+      - name: Build and analyze
+        uses: github/codeql-action/analyze@v2
```

This workflow is failing, might be due to how the steps are orchestrated, this could help (not entirely sure here though).
```suggestion
    steps:
      - name: Checkout code
        uses: actions/checkout@v3

      - name: Initialize CodeQL
        uses: github/codeql-action/init@v2
        with:
          languages: java

      - name: Setup Java
        uses: actions/setup-java@v3
        with:
          distribution: "corretto"
          java-version: "17"

      - name: Setup Gradle
        uses: gradle/gradle-build-action@v2.4.2
        with:
          gradle-version: 7.3
      
      - name: Execute build with Gradle wrapper
        run: ./gradlew build -PsonatypeUsername=$NEXUS_UNAME -PsonatypePassword=$NEXUS_PWD


      - name: Build and analyze
        uses: github/codeql-action/analyze@v2
```

---

Lets try this!